83-scrap-data-from-imdb.md

```diff
@@ -0,0 +1,40 @@
+# You are required to scrap data from IMDb top 250 movies page. It should only have fields movie name, year, and rating
+
+You need to use the following lines of code:
+
+```python
+from bs4 import BeautifulSoup
+import requests
+import sys
+
+url = 'http://www.imdb.com/chart/top'
+response = requests.get(url)
+soup = BeautifulSoup(response.text)
+tr = soup.findChildren("tr")
+tr = iter(tr)
+next(tr)
+
+for movie in tr:
+title = movie.find('td', {'class': 'titleColumn'} ).find('a').contents[0]
+year = movie.find('td', {'class': 'titleColumn'} ).find('span', {'class': 'secondaryInfo'}).contents[0]
+rating = movie.find('td', {'class': 'ratingColumn imdbRating'} ).find('strong').contents[0]
+row = title + ' - ' + year + ' ' + ' ' + rating
+
+print(row)
```

Comment:
Did not work for me:
Python 3.8.10 (default, Jun  2 2021, 10:49:15) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from bs4 import BeautifulSoup
>>> import requests
>>> import sys
>>> url = 'http://www.imdb.com/chart/top'
>>> response = requests.get(url)
>>> soup = BeautifulSoup(response.text)
>>> tr = soup.findChildren("tr")
>>> tr = iter(tr)
>>> next(tr)
<tr>
<th></th>
<th>Rank &amp; Title</th>
<th>IMDb Rating</th>
<th>Your Rating</th>
<th></th>
</tr>
>>> for movie in tr:
... title = movie.find('td', {'class': 'titleColumn'} ).find('a').contents[0]
  File "<stdin>", line 2
    title = movie.find('td', {'class': 'titleColumn'} ).find('a').contents[0]
    ^
IndentationError: expected an indented block
>>> year = movie.find('td', {'class': 'titleColumn'} ).find('span', {'class': 'secondaryInfo'}).contents[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'movie' is not defined
>>> rating = movie.find('td', {'class': 'ratingColumn imdbRating'} ).find('strong').contents[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'movie' is not defined
>>> row = title + ' - ' + year + ' ' + ' ' + rating
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'title' is not defined
>>> print(row)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'row' is not defined